stdlib/public/SDK/Foundation/JSONEncoder.swift

```diff
@@ -2453,3 +2456,18 @@ fileprivate extension EncodingError {
         return .invalidValue(value, EncodingError.Context(codingPath: codingPath, debugDescription: debugDescription))
     }
 }
+
+//===----------------------------------------------------------------------===//
+// Optionality Checker
+//===----------------------------------------------------------------------===//
+
+// This protocol is expected to be conformed to by Optional only.
+// It is used to check if a type is Optional.
+fileprivate protocol _Unwrappable {}
+
+extension Optional : _Unwrappable {}
+
+@inline(__always)
+fileprivate func _isOptionalType<T>(_ type: T.Type) -> Bool {
+    return type as? _Unwrappable.Type != nil
```

Comment:
This can be more easily written as `type is _Unwrappable.Type`, but looks reasonable to me otherwise overall.

Reply:
Fixed.